tox.ini

```diff
@@ -1,19 +1,13 @@
 [tox]
-envlist = py{27,35,36,37,38,39}
-
-[testenv:py27]
-setenv = EXCLUDE = --exclude=./.*,./examples/loop_asyncio.py,*/MQTTV5.py,*/MQTTV311.py
+envlist = py{27,37,38,39,310,311}
 
 [testenv]
-whitelist_externals = echo make
+# whitelist_externals = echo make
 deps =
-    -rrequirements.txt
-    flake8
+    # -rrequirements.txt
+    mock
+    pytest
 commands =
-    # $EXCLUDE is defined above in testenv:py27 as a workaround for Python 2
-    # which does not support asyncio and type hints
-    flake8 . --count --select=E9,F63,F7,F822,F823 --show-source --statistics {env:EXCLUDE:}
-    python setup.py test
-    make -C test test
```

Comment:
Those tests should be converted to be regular `py.test` tests IMO, they now seem to rely on some _very_ homespun code (e.g. some Python code in `.test` files that maybe somehow relates to the `.py` files...).

They're also not covered by any code coverage so long as that's happening.

Reply:
Agree, but they are not yet converted, so in meantime it seems better to kept them.

Reply:
Created #768 to track that.

Reply:
`make -C test test` might be useful but I was unable to make it work.

Calling `setup.py xxx` is deprecated because it was a great way to hack in malware.  So, let's allow pytest test discovery to find and run the tests (see below).

pytest can run old-style unit tests out-of-the-box so while converting tests might be useful, it is not obligatory.

`py.test` (with the dot) was deprecated years ago.

Reply:
PR #768 will re-introduce test lost from the make -C test